core/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/get/TransportGetSnapshotsAction.java

```diff
@@ -80,23 +80,22 @@ protected void masterOperation(final GetSnapshotsRequest request, ClusterState s
         try {
             final String repository = request.repository();
             List<SnapshotInfo> snapshotInfoBuilder = new ArrayList<>();
+            final Map<String, SnapshotId> allSnapshotIds = new HashMap<>();
+            final List<SnapshotId> currentSnapshotIds = new ArrayList<>();
+            for (SnapshotInfo snapshotInfo : snapshotsService.currentSnapshots(repository)) {
```

Comment:
if you set `_all` (i.e. `isAllSnapshots(request.snapshots()) == true`), you still want the current snapshots too.